TowersOfHanoi/TowersOfHanoi.cs

```diff
@@ -1,12 +1,150 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Collections;
+
 
 namespace TowersOfHanoi
 {
     class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            Game game = new Game();
+            game.drawBoard();
+            while(game.CheckForWin() != true)
+            {
+                Console.WriteLine("What tower to pick up from? (A, B, C)");
+                String pop = Console.ReadLine();
+                pop = pop.ToUpper();
+                Console.WriteLine("Where to set down the block?");
+                String push = Console.ReadLine();
+                push = push.ToUpper();
+                if(game.IsLegal(pop, push) == true)
+                {
+                    game.MovePiece(pop, push);
+                    game.drawBoard();
+                }
+                else{
+                    Console.WriteLine("That is not a legal move.");
+                }
+            }
+            Console.WriteLine("You won!!");
+            
+        }
+    }
+    class Block
+    {
+        public int Weight{get; private set;}   
+        public Block(int Weight)
+        {
+            this.Weight = Weight;
+        }
+    }
+    class Tower
+    {
+        public Stack<Block> Blocks{get;}
+        public Tower()
+        {
+            Blocks = new Stack<Block>();
+        }
+        public void Push(Block block)
+        {
+            Blocks.Push(block);
+        }
+    }
+    class Game
+    {
+		Dictionary<string, Tower> Towers;
+
+        public Game()
+        {
+            Towers = new Dictionary<String, Tower>();
+            Towers["A"] = new Tower();
+            Towers["B"] = new Tower();
+            Towers["C"] = new Tower();
+            Block a = new Block(1);
+            Block b = new Block(2);
+            Block c = new Block(3);
+            Block d = new Block(4);
+            Towers["A"].Push(a);
```

Comment:
When pushing items in the stack, it would have made sense to push the 4 then 3 then 2, then 1.